packages/augur-core/source/contracts/libraries/token/ERC777BaseToken.sol

```diff
@@ -0,0 +1,130 @@
+    uint256 public supply;
+
+    uint256 constant public granularity = 1;
+    uint8 constant public decimals = 18;
+
+    address[] internal initialDefaultOperators;
+    mapping(address => bool) internal isDefaultOperator;
+    mapping(address => mapping(address => bool)) internal revokedDefaultOperator;
+    mapping(address => mapping(address => bool)) internal authorizedOperators;
+
+    function initialize820InterfaceImplementations() internal returns (bool) {
+        setInterfaceImplementation("ERC777Token", this);
+        return true;
+    }
+
+    function totalSupply() public view returns (uint256) {
+        return supply;
+    }
+
+    function balanceOf(address _tokenHolder) public view returns (uint256) {
+        return balances[_tokenHolder];
+    }
+
+    function defaultOperators() public view returns (address[] memory) {
+        return initialDefaultOperators;
+    }
+
+    function send(address _to, uint256 _amount, bytes32 _data) public returns (bool) {
+        doSend(msg.sender, msg.sender, _to, _amount, _data, "", true);
+        return true;
+    }
+
+    function authorizeOperator(address _operator) public returns (bool) {
+        require(_operator != msg.sender, "Cannot authorize yourself as an operator");
+
+        if (isDefaultOperator[_operator]) {
+            revokedDefaultOperator[_operator][msg.sender] = false;
+        } else {
+            authorizedOperators[_operator][msg.sender] = true;
+        }
+        emit AuthorizedOperator(_operator, msg.sender);
+        return true;
+    }
+
+    function revokeOperator(address _operator) public returns (bool) {
+        require(_operator != msg.sender, "Cannot revoke yourself as an operator");
+
+        if (isDefaultOperator[_operator]) {
+            revokedDefaultOperator[_operator][msg.sender] = true;
+        } else {
+            authorizedOperators[_operator][msg.sender] = false;
+        }
+
+        emit RevokedOperator(_operator, msg.sender);
+        return true;
+    }
+
+    function isOperatorFor(address _operator, address _tokenHolder) public view returns (bool) {
+        return (_operator == _tokenHolder // solium-disable-line operator-whitespace
+            || authorizedOperators[_operator][_tokenHolder]
+            || (isDefaultOperator[_operator] && !revokedDefaultOperator[_operator][_tokenHolder]));
+    }
+
+    function operatorSend(address _from, address _to, uint256 _amount, bytes32 _data, bytes32 _operatorData) public returns (bool) {
+        require(isOperatorFor(msg.sender, _from), "Not an operator");
+        doSend(msg.sender, _from, _to, _amount, _data, _operatorData, true);
+        return true;
+    }
+
+    function doSend(address _operator, address _from, address _to, uint256 _amount, bytes32 _data, bytes32 _operatorData, bool _preventLocking) internal returns (bool) {
+        callSender(_operator, _from, _to, _amount, _data, _operatorData);
+
+        require(_to != address(0), "Cannot send to 0x0");
```

Comment:
This will need to be considered carefully. If a user sets the market creator to 0x0, they could shutdown all trading and claiming on a market (with no way to fix) (this is true with the removal of mailbox). Previous tokens have not had 0x0 restrictions so this was not a concern.

Reply:
That's a great point. Will need to review for any user specified addresses in the system.

Reply:
On the plus side, inspecting log transfers to/from 0x0 will no longer be ambiguous